src/ds/ds-options.cpp

```diff
@@ -33,7 +33,7 @@ namespace librealsense
         }
     }
 
-    emitter_option::emitter_option(uvc_sensor& ep)
+    emitter_option::emitter_option( const std::weak_ptr< uvc_sensor > & ep )
```

Comment:
Is it ok to pass a shared_ptr to a weak_ptr reference?
Why reference and not the object itself?

Reply:
Yes, a `shared_ptr` can be passed to a `weak_ptr`. https://en.cppreference.com/w/cpp/memory/weak_ptr/weak_ptr.
Using a reference is more efficient - no temp `shared_ptr`s updating reference count

Reply:
OK Thanks